src/main/java/uk/gov/ons/ctp/response/action/endpoint/ActionEndpoint.java

```diff
@@ -243,6 +245,35 @@ public ActionDTO updateAction(
     }
   }
 
+  /**
+   * PUT to re-run the aborted Actions for a specified actionId and to reschedule them .
+   *
+   * @param actionIds Action Id of the Action/s to re run
+   * @return Success for the re ran actions
+   * @throws CTPException if update operation fails
+   */
+  @RequestMapping(value = "/rerun", method = RequestMethod.PUT)
+  public ResponseEntity<List<UUID>> rerunAction(
+      @RequestParam(value = "actionIds") List<UUID> actionIds) throws CTPException {
+    log.with("action_id", actionIds).info("Re-scheduling Aborted Action");
+
+    for (UUID action : actionIds) {
```

Comment:
The problem with having the loop here rather than in the service class, is that there's no transaction demarkation here. In the event that _some_ of the actionIds succeed, then one fails, those changes will already have been persisted to the database but the request can't be made for a second time, because then it'd be trying to transition actions from the state they're already in.

You should move as much logic as possible into the service class and out of the endpoint class.